### PR TITLE
Links list proposal

### PR DESCRIPTION
**Pull request by gengel**
A proposal for embedded and curated "See More" boxes

### CODE REVIEW

**Comment by gengel** [comment]
Temporarily rescinded -- this needs more changes

**Comment by gengel** [comment]
Reopened!

File: docs/proposals/2019-03-18-Links-List.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      "subtype": "sidebar",
+      "title": "More 'Biting' Stories",
+
+      "items": [
+        {
+          "type": "links_list_item",
+          "subtype": "blue-label",
+
+          "title": "Lorem ipsum",
+          "link": "http://www.google.com",
```
Comment (chenghsinw) [inline docs/proposals/2019-03-18-Links-List.md]: From the last email for Prisa, looks like they need a way to configure additional data for the link: rel="nofollow" and target="_blank". Do u think we should have additional_propertties in links_list_item for custom requirements or introduce specific properties for the needs?

Comment (gengel) [reply docs/proposals/2019-03-18-Links-List.md]: I think they should key off of subtype if they really need it.

Comment (yury-sannikov) [reply docs/proposals/2019-03-18-Links-List.md]: It seems it really depends on how many parameters users want to control. For instance, for RED/BLUE/BLACK label color and nofollow/blank settings, we will have to provide 12 subtypes.

Comment (chadmoone) [reply docs/proposals/2019-03-18-Links-List.md]: Are you suggesting this be a text item which can have an `<a>` tag? I think that opens this up far more than we want to.

Comment (chenghsinw) [reply docs/proposals/2019-03-18-Links-List.md]: Do u mean they have to define subtypes something like these:
- blue-label-nofollow
- blue-label
- red-label-nofollow
- red-label
- green-label-nofollow
- green-label
- ...
It looks like more complex and each org might implement the pattern differently which will make Ellipsis hard to standardize the UI.

Comment (gengel) [reply docs/proposals/2019-03-18-Links-List.md]: @chadmoone No tags. The entire entry is a link to the URL specified.

Comment (gengel) [reply docs/proposals/2019-03-18-Links-List.md]: @chenghsinw @yury-sannikov We should not assume a specific rendering. We're trying to capture a general version of the use case. The front-end can key off of the subtype to determine labels, colors, nofollow, whatever other presentation options they want.  If there is a proven widespread demand for labels or nofollow, specifically, we can add those options at a later time.

Comment (StephanieDClark) [inline docs/proposals/2019-03-18-Links-List.md]: @gengel This looks alot like "interstitial_link".  Would this be intended as a replacement, with a new subtype of "interstitial_link"?

Comment (gengel) [reply docs/proposals/2019-03-18-Links-List.md]: It's a similar use case, but not quite the same. The intention is that this is it's own, new element type for groups of links with a title for the box/section, a plaintext title for each link, optional images, but no embedded HTML or formatting. 

This is a little bit more structured, and more of an "aside" to the main content.

**Comment by scarothers** [comment]
Agree about link_list as the name. Overall makes sense to me – I've seen a lot of clients want to curate related stories within the body of the article (specifically able to place them at a certain point in the body of the article, which you can't do with related_content), and I think this makes sense for that use case. Two questions: 

* With title being plaintext, does that mean it couldn't include italics? I think people might need that if the headline has an italicized word in it ... should it just be the same as whatever headlines.basic accepts? 
* I agree/understand about this not needing to be a story reference (because it makes the payload so big). Sometimes clients do want to display data about the related story though – like section or date. Should that just be considered something that's not supported by *links_list*?

File: docs/proposals/2019-03-18-Links-List.md (hunk 1 above)
Comment (gengel) [reply docs/proposals/2019-03-18-Links-List.md]: We did call the link "url" over in interstitial, though, so I should probably change the field here to be called the same.

**Comment by gengel** [comment]
@scarothers As described here, this proposal would not cover italics or additional metadata. 

Italics is probably desirable, but it's hard to capture safely without supporting all of #134

**Comment by scarothers** [comment]
@gengel makes sense. if clients usually use italics in headline style, we can suggest they use single quote marks instead here.

File: docs/proposals/2019-03-18-Links-List.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  "headlines": {
+    "basic": "Man Bites Dog"
+  },
+
+  "content_elements": [
+    {
+      "type": "link_list",
+
+      "subtype": "sidebar",
+      "title": "More 'Biting' Stories",
```
Comment (StephanieDClark) [inline docs/proposals/2019-03-18-Links-List.md]: I don't think the title for the type or the items should be required.  

Use cases I see for this include: 
- support the use of a "read more" list. 
In this case, the set of links has a title, "Read More:", but the individual links don't.  See https://www.washingtonpost.com/opinions/2019/03/25/trump-won-with-illicit-help-he-abused-his-power-his-ag-is-blocking-full-reckoning/?utm_term=.cfcdb102bdb2  

- replace the use of content element type "interstitial_link".  
In this case, in interstitial link is really a set of 1, with no headlines.

Comment (gengel) [reply docs/proposals/2019-03-18-Links-List.md]: What would the link text be if not the title?

Comment (robpc) [reply docs/proposals/2019-03-18-Links-List.md]: I think maybe what @StephanieDClark is getting at here is there are examples where there is supporting text for a link that is not the link itself? I know that historically this has been available (especially with interstitial links) but was not able to find a recent example.

Comment (RyanGladstone) [reply docs/proposals/2019-03-18-Links-List.md]: From the journalist's perspective, the interstitial_link content element requires fewer steps than using a links list with a _interstitial_link_ subtype and no headline

**Comment by st421** [comment]
I find this approach confusing. As I mentioned over [here](https://github.com/washingtonpost/ans-schema/pull/201#issuecomment-476356704), people want to add links to single items and I think there should be a way to do so without putting the items in a `link_list`. If elements had a field where you could not only add a link but also specify if the link was to (1) a place internal to the current document (2) another ANS document (so a canonical URL can be used) (3) anywhere else, then it seems like an element group would work when you need a group of linked items, but you'd also be able to distribute your linked items throughout a story if you wanted to (okay, the `element_group` type lacks a `title` but probably it should have one either way).

**Comment by gengel** [comment]
Per feedback from @st421 and @StephanieDClark, I'm going to revisit this to make the relationship between `link_list` and `interstitial_link` more explicit

**Comment by gengel** [comment]
@st421 @StephanieDClark @styonsk @jmjakes @yury-sannikov 

Re-opened. Now uses `interstitial_link` and extends that element.

**Pull request merged**